modules/home-manager/zsh-syntax-highlighting.nix

```
@@ -1,18 +1,51 @@
+# In this file we define both `catppuccin.zsh-syntax-highlighting` and `programs.zsh.syntaxHighlighting.catppuccin` options this is beacuse
+# we have to recreate the lib.mkRenamedOptionModule here due to the use of a submodule being used for `programs.zsh.syntaxHighlighting`
+# which would lead to an error if we used the original `lib.mkRenamedOptionModule`.
+#  see <https://github.com/catppuccin/nix/pull/371#discussion_r1835785145>
 { config, lib, ... }:
 let
   inherit (config.catppuccin) sources;
   inherit (lib) ctp;
-  cfg = config.programs.zsh.syntaxHighlighting.catppuccin;
+  cfg = config.catppuccin.zsh-syntax-highlighting;
   enable = cfg.enable && config.programs.zsh.syntaxHighlighting.enable;
+
+  oldCfg = config.programs.zsh.syntaxHighlighting.catppuccin;
 in
 {
-  options.programs.zsh.syntaxHighlighting.catppuccin = ctp.mkCatppuccinOpt {
-    name = "Zsh Syntax Highlighting";
-  };
+  options = {
+    catppuccin.zsh-syntax-highlighting = ctp.mkCatppuccinOpt {
+      name = "Zsh Syntax Highlighting";
+      enableDefault = config.catppuccin.enable || oldCfg.enable;
+    };
 
-  config.programs.zsh = lib.mkIf enable {
-    initExtra = lib.mkBefore ''
-      source '${sources.zsh-syntax-highlighting}/themes/catppuccin_${cfg.flavor}-zsh-syntax-highlighting.zsh'
-    '';
+    programs.zsh.syntaxHighlighting.catppuccin =
+      lib.recursiveUpdate
+        (ctp.mkCatppuccinOpt {
+          name = "Zsh Syntax Highlighting";
+          enableDefault = false;
+        })
+        # hide the options from the search
+        {
+          enable.internal = true;
+          flavor.internal = true;
+        };
```

```suggestion
    programs.zsh.syntaxHighlighting = lib.mkOption {
      type = lib.types.submodule {
        options.catppuccin =
          ctp.mkCatppuccinOpt {
            name = "Zsh Syntax Highlighting";
            enableDefault = false;
          }
          // {
            # hide the options from the search
            enable.internal = true;
            flavor.internal = true;
          };
      };
    };
```
We should also be able to extend the submodule like so. I don't *think* it makes much difference here, but just in case anything breaks